### ISSUE TITLE
Wired message in 8.5.2

### ISSUE THREAD

**Issue by donho** [notepad-plus-plus/nppShell#29]
`SYSPRP Package NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6 was installed for a user, but not provisioned for all users.`
https://community.notepad-plus-plus.org/topic/24345/notepad-v8-5-2-release/19?_=1681357704510

Not sure if it’s NppShell issue though.
What do you think @GurliGebis ?

**Comment by GurliGebis** [comment]
It is the same as when we try to install it for all users.
Sysprep complains that the package isn't installed for all users - so when the image they make is installed, the new user will not be have the package installed.

However, the multi-user fix we did earlier this week also works there. So the first time they right click a file after they create the user, it will be there.

This requires 8.5.3 though, so the version they are using (8.5.2) doesn't do that 🙂

**Comment by donho** [comment]
@GurliGebis 
> This requires 8.5.3 though, so the version they are using (8.5.2) doesn't do that 

Sorry I don't follow you - what's in v8.5.3 makes difference?

> It is the same as when we try to install it for all users.
> Sysprep complains that the package isn't installed for all users - so when the image they make is installed, the new user will not be have the package installed.

So if I understand correctly, while installer registering NppShell.dll for the Adm user, Sysprep has detected it and complains? If it is, Sysprep will always complain with our current solution (installing for user on the first use), won't it?

**Comment by GurliGebis** [comment]
@donho in the test builds we have now, this code exists: https://github.com/notepad-plus-plus/nppShell/blob/e250ff9af44281cbff23b9d246cf06974f232cef/Installer.cpp#L409

Sysprep complains that the package is only installed for a normal user, and not provisioned for everyone (which is what you cannot do for a sparse package - or to be more specific, you can do it, but it doesn't get installed for other users when they log in).
So since the code I link above exists, they should just ignore the warning from sysprep, since we handle registering the package for new users at the first right click on a file.

**Comment by donho** [comment]
@GurliGebis 
> So since the code I link above exists, they should just ignore the warning from sysprep, since we handle registering the package for new users at the first right click on a file.

So should [TEST20](http://download.notepad-plus-plus.org/repository/MISC/nppShell.TEST20/) fix it?

**Comment by GurliGebis** [comment]
@donho it will make sure it is installed for the user on first right click.
It won't prevent sysprep from showing that warning, since it will complain about it, but the warning can be safely ignore, since it is working as intended

**Comment by donho** [comment]
@GurliGebis 
> it will make sure it is installed for the user on first right click.

You mean on the **2nd** right click, right?

> It won't prevent sysprep from showing that warning, since it will complain about it, but the warning can be safely ignore, since it is working as intended

OK, thank you for your answer.

**Comment by GurliGebis** [comment]
@donho on the first right click, it will trigger the install of the msix package, so it will be there on the second right click 🙂

**Comment by donho** [comment]
@GurliGebis 
> @donho on the first right click, it will trigger the install of the msix package, so it will be there on the second right click 🙂

Yes, sure. Sorry for misreading. Thank you for the reply.

**Comment by GurliGebis** [comment]
🙂

**Comment by Hammerfest** [comment]
Thanks everyone for reporting this error, I was just building new images for family systems updating to 11 and kept running into this error when trying to sysprep when it worked without issue a few months ago!

Looks like it will be resolved in 8.5.3 so I will keep an eye out and test when it drops!

**Comment by GurliGebis** [comment]
@Hammerfest sysprep fails if a package is provisioned for the user but not the system?

**Comment by Hammerfest** [comment]
> 

Correct, exact message at the opening comment

**Comment by GurliGebis** [comment]
@Hammerfest can you try this, and then run sysprep again:
`Get-AppxPackage -Name *NotepadPlusPlus* | Remove-AppxPackage`

**Comment by GurliGebis** [comment]
The package will be reinstalled automatically

**Comment by Hammerfest** [comment]
> 

This worked, it should be noted however that if you create a template user (not done often by many) it will need to be done on both the user and sysprep account or it will still fail.

Might be to much of a PITA to properly resolve this to simply install upon account creation or provide an integration pack like Microsoft does for some of its tools.

**Comment by GurliGebis** [comment]
@Hammerfest the problem is that provisioning the package on a system level doesn't work for sparse package, so we simply cannot do that.
For it to work, the entire NPP setup should be moved into an MSIX package instead, but that is not happening right now at least.

**Comment by levicki** [comment]
@GurliGebis
> It won't prevent sysprep from showing that warning, since it will complain about it, **but the warning can be safely ignore**, since it is working as intended

I don't know with what authority you are saying this, but it is totally incorrect.

Sysprep is not issuing a warning which can be ignored &mdash; it is throwing an error since it fails to reseal the image to OOBE state because of this improperly provisioned package.

When we are at the subject of that package, I don't like how it is done silently without notifying the user that it is being installed. Such behavior is always automatically associated in my head with malware.

I don't know what is the purpose of that package (and frankly I don't care because it seems to work for me without it), but if it interferes with `sysprep` it will prevent staging Notepad++ in a Windows image and leave many people scratching their head as to where it came from.

Changes like this should be properly tested or at least announced clearly in release notes &mdash; you can't expect people to come read the commit logs and merge / pull requests to find out stuff like that.

**Comment by donho** [comment]
@GurliGebis 
Is it possible (doable) to integrate the solution suggested by Luper?
https://community.notepad-plus-plus.org/topic/24591/notepad-v8-5-4-release/4?_=1688091290446

**Comment by levicki** [comment]
@donho I don't think that's possible, @GurliGebis said sparse package cannot be provisioned on the system level.

It's much easier to do this using the Active Setup registry key instead of building an appx as I have explained in the post above yours.

**Comment by GurliGebis** [comment]
@donho from what I tried earlier, when we had to find a solution for installing it for all users on the system, it seems like it just doesn't work.
It registers the package just fine, but it never gets installed for any users.
It might be me doing something wrong, since I wasn't able to find any documentation or examples (or other people on github doing it) - it's difficult to know 🙂
That is way we have to do it this way (which is sub-optimal, but at least works).
The only problem is for people using sysprep, since they have to unregister the package before sysprepping the installation. (It will be registered for each user on first right click like as usual afterwards)

**Comment by Neustradamus** [comment]
To follow

**Comment by rgb99** [comment]
> need to be done on both the user and sysprep account

@Hammerfest How, exactly? I installed NPP as Administrator in Audit Mode -- there are no other accounts. Running `Get-AppxPackage -Name *NotepadPlusPlus* | Remove-AppxPackage` works after installing 8.5.2, but when doing the same thing on newer versions of NPP, sysprep failed following Remove-AppxPackage.

**Comment by Hammerfest** [comment]
> 

I haven't tried with anything past 8.5.3, its possible something changed
When I mentioned user I meant if you made a template user account (something I do so layouts, theme settings stick to any new user created), you can ignore that if you did not/do not do this.

Outside of that the process was reboot into audit mode from OOBE, install programs, restart, run command, then perform sysprep shutdown to OOBE, I wanna say if you restart at all after running the command to remove the package it re-installs on next login of the account

I need to make a new image, will try that this weekend.

**Comment by rgb99** [comment]
@Hammerfest Thanks for the quick reply! In my case, it's for VMware Horizon (VDI), non-persistent desktops. I can confirm that the AppxPackage comes back during the next logon, but in my experience, removing the package and then running sysprep immediately after (on NPP 8.5.3 and newer) failed.

When we update the Desktop Pools later this month, the machine will have NPP 8.5.2 on it. NPP is already up to 8.5.5 and the sysprep issue is not resolved. Hopefully, it will be, sooner than later!

**Comment by benlye** [comment]
> @Hammerfest Thanks for the quick reply! In my case, it's for VMware Horizon (VDI), non-persistent desktops. I can confirm that the AppxPackage comes back during the next logon, but in my experience, removing the package and then running sysprep immediately after (on NPP 8.5.3 and newer) failed.

I'm also building images for a VMware Horizon VDI environment. I have an automated build process which builds and deploys our golden image templates. I just checked last night's build and it worked fine with NPP 8.5.4 in the image.

The build uses Chocolatey to install NPP and then removes the AppX pacakge:
```
choco install notepadplusplus.install -y
Get-AppxPackage NotepadPlusPlus | Remove-AppXPackage
```

The Horizon customization process ran fine afterwards.

**Comment by rgb99** [comment]
@benlye That is great to know! It's a manual process for us, so we'll have to try 8.5.4 next month.

**Comment by rgb99** [comment]
Notepad++ 8.5.7 addresses some bad vulnerabilities. However, after running the Remove-AppxPackage command, sysprep still breaks. Is there a way to fix this? @benlye , have you come across this?

**Comment by benlye** [comment]
> Notepad++ 8.5.7 addresses some bad vulnerabilities. However, after running the Remove-AppxPackage command, sysprep still breaks. Is there a way to fix this? @benlye , have you come across this?

My build process appears to be working fine with NPP 8.5.7 using the `Remove-AppXPackage` mitigation.

**Comment by rgb99** [comment]
@benlye Lucky you! I rebooted the image. After a few minutes, the package came back so I removed it again with the command, and then Generalize via OSOT worked. I don't understand... but I'll take it.

**Comment by AlonzoMoselyFBI** [comment]
> @Hammerfest can you try this, and then run sysprep again: `Get-AppxPackage -Name *NotepadPlusPlus* | Remove-AppxPackage`

I created a github account just to thank you for this - fixed my issue.  Thank you!

**Comment by brentil** [comment]
I wanted to report we're seeing this still with 8.6.4 as we're building our first Windows 11 sysprep image.  I can run the command above but I have to immediately sysprep the system otherwise it will re-inject the Appx within minutes.  Would it be possible to when the application is installed as admin to provision the Appx to all users instead of just the installed one?  That would solve this issue.  Or provide the ability to add a flag to disable this behavior for image based systems?

The extra frustrating part is we've disable the new Windows 11 right click menu to use the Windows 10 style one so the Appx package isn't even needed.

**Comment by GurliGebis** [comment]
@brentil That does not work for some reason.
We have tried pretty much everything - the API calls return just fine, and shows the package as provisioned for all users - but no users get it installed when they log in (not even the user that does the provisioning for all users).
I would suspect this is due to the package not being from the store, but I can only guess.

**Comment by OrionWithrow-OHIT** [comment]
Still struggling with this on W11 23H2, using NPP 8.6.7. Was working at one point with a previous version.

**Comment by BrendenWalker** [comment]
Same problem here with 8.6.7 trying to syspred an Azure VM.  

Running `Get-AppxPackage | findstr /I "note"`  from admin PS returns no notepad++ packages installed.

Tried various versions of Remove-AppxPackage including running for each user.  

Deleted all remants in registry and on local filesystem, and still getting:

`2024-06-06 17:39:39, Error                 SYSPRP Package NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6 was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image.`

I got this to work yesterday on a VM created via the same snapshot from the same VM source... so I'm not sure what the heck is going on...  Just adding my notes to this in case there's anything useful.

**Comment by OrionWithrow-OHIT** [comment]
Older versions still work, see my notes above.

[signature_1901198342]<https://optimumhit.com/>



Orion Withrow (He/Him/His)
Cloud Architect

Phone: 804-388-2971
Email: ***@***.******@***.***>
www.optimumhit.com<https://www.optimumhit.com/>

TRULY HUMAN NOTICE: Getting this email out of normal working hours? We work at a digitally enabled relentless pace, which can disrupt our ability to sleep enough, eat right, exercise, and spend time with the people that matter most. I am sending you this email at a time that works for me. Please respond to it at a time that is convenient for you.


From: Brenden Walker ***@***.***>
Date: Thursday, June 6, 2024 at 1:55 PM
To: notepad-plus-plus/nppShell ***@***.***>
Cc: Orion Withrow ***@***.***>, Comment ***@***.***>
Subject: Re: [notepad-plus-plus/nppShell] Sysprep warning message in 8.5.2 (Issue #29)
CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.


Same problem here with 8.6.7 trying to syspred an Azure VM.

Running Get-AppxPackage | findstr /I "note" from admin PS returns no notepad++ packages installed.

Tried various versions of Remove-AppxPackage including running for each user.

Deleted all remants in registry and on local filesystem, and still getting:

2024-06-06 17:39:39, Error SYSPRP Package NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6 was installed for a user, but not provisioned for all users. This package will not function properly in the sysprep image.

I got this to work yesterday on a VM created via the same snapshot from the same VM source... so I'm not sure what the heck is going on... Just adding my notes to this in case there's anything useful.

**Comment by jlsupremo1** [comment]
Yes, this happened after the option to edit using the reverse button was included in Windows. Versions that did not have this option are free from this problem with sysprep. Can the programmers help make a version with this fix? I really need Notepadd++ on my versions of Windows using sysprep, but I'm having to use the npp.8.5.Installer.x64.exe version that did not have this option yet. If I install the new version, this problem occurs.
I just tried with Notepad++ v8.7 on Windows 11 24H2, still the same problem when generalizing sysprep it presents the error, canceling the windows sysprep.

**Comment by brentil** [comment]
> I really need Notepadd++ on my versions of Windows using sysprep, but I'm having to use the npp.8.5.Installer.x64.exe version that did not have this option yet. If I install the new version, this problem occurs. I just tried with Notepad++ v8.7 on Windows 11 24H2, still the same problem when generalizing sysprep it presents the error, canceling the windows sysprep.
> 
@jlsupremo1 all you have to do to fix it with a newer version of Notepad++ is to remove the Appx right before you call sysprep.  Just don't right click on anything between that and running sysprep because it'll re-inject it again...

Get-AppxPackage -Name *NotepadPlusPlus* | Remove-AppXPackage

**Comment by jlsupremo1** [comment]
I didn't understand, but removing this (Get-AppxPackage -Name NotepadPlusPlus | Remove-AppXPackage) doesn't affect notepad++ after reconfiguring windows? Isn't it missing a component or itself?


> > I really need Notepadd++ on my versions of Windows using sysprep, but I'm having to use the npp.8.5.Installer.x64.exe version that did not have this option yet. If I install the new version, this problem occurs. I just tried with Notepad++ v8.7 on Windows 11 24H2, still the same problem when generalizing sysprep it presents the error, canceling the windows sysprep.
> 
> @jlsupremo1 all you have to do to fix it with a newer version of Notepad++ is to remove the Appx right before you call sysprep. Just don't right click on anything between that and running sysprep because it'll re-inject it again...
> 
> Get-AppxPackage -Name _NotepadPlusPlus_ | Remove-AppXPackage

**Comment by brentil** [comment]
No, all it's doing is removing the Windows 11 right click injection application, it's not Notepad++ you're removing.  That injection app is what's causing the sysprep issues and not Notepad++ itself.

**Comment by oranmizrahi** [comment]
So what to do to fix this problem?

**Comment by GurliGebis** [comment]
@oranmizrahi run the Remove-AppXPackage, followed by sysprepping the image.

**Comment by oranmizrahi** [comment]
> @oranmizrahi run the Remove-AppXPackage, followed by sysprepping the image.

I already ran Remove-AppxPackage and attempted to Sysprep the image, but it didn't work. Notepad++ still shows up when checking with Get-AppxPackage -AllUsers -Name *NotepadPlusPlus*.

**Comment by GurliGebis** [comment]
Then restart, and it will be removed - just don't right click in explorer, since that will trigger the reinstall.

**Comment by oranmizrahi** [comment]
> Then restart, and it will be removed - just don't right click in explorer, since that will trigger the reinstall.
Thank you, it really worked! Notepad wasn’t the only package causing issues.

If someone else encounter problems with residual Appx packages, follow these steps to ensure complete removal:

1. Check if the Package Exists: Use Get-AppxPackage -AllUsers -Name PackageNameChangeMe.
2. Remove the Package: Use Get-AppxPackage -AllUsers -Name *PackageNameChangeMe* | Remove-AppxPackage -AllUsers.
3. Remove the Provisioned Package: Use Get-AppxProvisionedPackage -Online | Where-Object {$_.DisplayName -like "*PackageNameChangeMe*"} | Remove-AppxProvisionedPackage -Online

**Comment by DamienDennehy** [comment]
This issue is also present in Windows Server 2025.

**Comment by brentil** [comment]
Could we maybe get a command line switch for the installer to disable the installation of AppxPackage?

**Comment by patrix87** [comment]
Running : `Reset-AppxPackage -Package "NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6"` seems to resolve the issue on Windows Server 2025.

**Comment by dBmQuake** [comment]
> Running : `Reset-AppxPackage -Package "NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6"` seems to resolve the issue on Windows Server 2025.

This command worked for me under Win11 Pro 24H2 26100.3194, Notepad++ v8.7.7.      
Notepad++ remains installed after the sysprep image is deployed.     
Thank you!

**Comment by ankurvvvv** [comment]
> Running : `Reset-AppxPackage -Package "NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6"` seems to resolve the issue on Windows Server 2025.

Love this. Worked on windows 11 Ent latest build.

**Comment by gloatoriginal** [comment]
> Running : `Reset-AppxPackage -Package "NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6"` seems to resolve the issue on Windows Server 2025.

This worked for me aswell on multi-tenant windows 11, shortened to oneliner without the need to specify the full string of the package:
`Get-AppxPackage -Name *NotepadPlusPlus* | Reset-AppxPackage`

**Comment by pier62350** [comment]
For information the package name changed in version 8.8.7 (maybe because it is now signed by Digicert)
the new package name is "NotepadPlusPlus_1.0.0.0_neutral__2247w0b46hfww"

So you can now run:

`Reset-AppxPackage -Package "NotepadPlusPlus_1.0.0.0_neutral__2247w0b46hfww"`
The best option is not using the full name as suggested above:

` Get-AppxPackage -Name *NotepadPlusPlus* | Reset-AppxPackage`

**Comment by musklor** [comment]
Hi, no more issue with the msi version.

**Comment by Neustradamus** [comment]
Have you progressed on it?

Thanks in advance.

Linked to:
- https://github.com/notepad-plus-plus/nppShell/issues/29
- https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13675
- https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13839
- https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13775

**Comment by dhs-rec** [comment]
When using recent versions, things work fine up to 8.8.6. Versions 8.8.7 and later break sysprep again.
